src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -75,7 +76,30 @@ const defaultProps = {
 
 class ReimbursementAccountPage extends React.Component {
     componentDidMount() {
-        fetchFreePlanVerifiedBankAccount();
+        // We can specify a specific step to navigate to by using route params when the component mounts.
+        fetchFreePlanVerifiedBankAccount(this.getStepToOpenFromRouteParams());
+    }
+
+    componentDidUpdate(prevProps) {
+        const currentStep = lodashGet(
+            this.props,
+            'reimbursementAccount.achData.currentStep',
+            CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT,
+        );
+        const previousStep = lodashGet(
+            prevProps,
+            'reimbursementAccount.achData.currentStep',
+            CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT,
+        );
+
+        if (currentStep === previousStep) {
+            return;
+        }
+
+        // When the step changes we will navigate to update the route params. This is mostly cosmestic as we only use
+        // the route params when the component first mounts to jump to a specific route instead of picking up where the
+        // user left off in the flow.
```

Comment:
```suggestion
        // When the step changes we will navigate to update the route params. This is mostly cosmetic as we only use
        // the route params when the component first mounts to jump to a specific route instead of picking up where the
        // user left off in the flow.
```